packages/cardano-services/test/ChainHistory/DbSyncChainHistoryProvider/ChainHistoryBuilder.test.ts

```diff
@@ -0,0 +1,217 @@
+import { Cardano } from '@cardano-sdk/core';
+import { ChainHistoryBuilder } from '../../../src';
+import { Pool } from 'pg';
+
+describe('ChainHistoryBuilder', () => {
+  let dbConnection: Pool;
+  let builder: ChainHistoryBuilder;
+
+  beforeAll(async () => {
+    dbConnection = new Pool({ connectionString: process.env.DB_CONNECTION_STRING });
+    builder = new ChainHistoryBuilder(dbConnection);
+  });
+
+  afterAll(async () => {
+    await dbConnection.end();
+  });
+
+  describe('queryTransactionInputsByHashes', () => {
+    test('query transaction inputs by tx hashes', async () => {
+      const result = await builder.queryTransactionInputsByHashes([
+        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819'),
+        Cardano.TransactionId('952dfa431223fd671c5e9e048e016f70fcebd9e41fcb726969415ff692736eeb')
+      ]);
+      expect(result).toHaveLength(2);
+      expect(result).toMatchSnapshot();
+    });
+    test('query transaction inputs with empty array', async () => {
+      const result = await builder.queryTransactionInputsByHashes([]);
+      expect(result).toHaveLength(0);
+    });
+    test('query transaction inputs when tx hashes not found', async () => {
+      const result = await builder.queryTransactionInputsByHashes([
+        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000')
+      ]);
+      expect(result).toHaveLength(0);
+    });
+
```

Comment:
This would fail in the `Cardano.TransactionId` function, which is tested [here](https://github.com/input-output-hk/cardano-js-sdk/blob/dc33f153288255b256453209433f08dae1a22291/packages/core/test/Cardano/types/Transaction.test.ts). It would be best to branch off _master_, and include it there.

Reply:
LGTM